src/magellan_firmware/firmware/magellan_controller/MagellanIMU.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * Teensy IMU for PittRAS Magellan
+ * Adapted from PittRAS Sailbot 
+ * Original Authors: Andrew Lobos and Kaylene Stocking
+ * Modified by: Xinke Chen
+ * 2018-10-03
+ */
+#ifndef MAGELLAN_IMU_H_
+#define MAGELLAN_IMU_H_
+
+#include <ros.h>
+#include <Adafruit_BNO055.h>
+#include <sensor_msgs/Imu.h>
+#include <geometry_msgs/Quaternion.h>
+#include <magellan_core/IMUState.h>
+
+class MagellanIMU{
+public:
+    MagellanIMU(ros::NodeHandle &imu_handle);
+    double getHeading();
+    void update();
+private:
+    ros::NodeHandle node_handle;
```

Comment:
We're not being ultra-strict about following the Google C++ style, but I would like to be consistant at least on variable naming (particularly local/class members): https://google.github.io/styleguide/cppguide.html#Variable_Names

Also sorry Arduino code isn't linted :(